### PR TITLE
Bump drf-spectacular from 0.13.2 to 0.14.0

### PR DESCRIPTION
**Pull request by fao89**
[noissue]

Please be sure you have read our documentation on creating PRs:
https://docs.pulpproject.org/contributing/pull-request-walkthrough.html

### CODE REVIEW

File: pulpcore/openapi/__init__.py [hunk 1]
```diff
@@ -237,9 +237,9 @@ def _resolve_path_parameters(self, variables):
 
         return parameters
 
-    def _map_serializer_field(self, field, direction):
+    def _map_serializer_field(self, *args, **kwargs):
         """Map serializer field."""
-        mapped = super()._map_serializer_field(field, direction)
+        mapped = super()._map_serializer_field(*args, **kwargs)
         if "additionalProperties" in str(mapped):
             del mapped["additionalProperties"]
```
Comment (mdellweg) [inline pulpcore/openapi/__init__.py:244]:
```suggestion
        mapped.pop("additionalProperties", None)
```

File: pulpcore/openapi/__init__.py [hunk 2]
```diff
@@ -237,9 +237,9 @@ def _resolve_path_parameters(self, variables):
 
         return parameters
 
-    def _map_serializer_field(self, field, direction):
+    def _map_serializer_field(self, *args, **kwargs):
```
Comment (mdellweg) [inline pulpcore/openapi/__init__.py:240]: Would you mind explaining, why we are patching this method?
Even reading the openapi docs, I don't really understand what the additionalProperties are for. Maybe you know, and why pulp does not want it.

Comment (fao89) [reply pulpcore/openapi/__init__.py:240]: I don't remember why, and it came from https://github.com/pulp/pulpcore/commit/c42931f30bc5320c763418abd86462b592e7b089
I'll remove it and run some tests to understand why and then document it

Comment (fao89) [reply pulpcore/openapi/__init__.py:240]: I still don't know exactly what it does, but it breaks some bindings tests

```
> /home/vagrant/devel/pulp_ansible/pulp_ansible/tests/functional/api/collection/v3/test_serializers.py(82)test_v3_collection_version_from_synced_data()
-> version = self.collections_versions_v3api.read(
(Pdb) self.collections_versions_v3api.read("nxos", "cisco", distribution.base_path, "1.4.0")
*** AttributeError: 'list' object has no attribute 'items'

pulp_ansible/tests/functional/api/collection/v3/test_serializers.py:82:                                                                                                                       
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _                                                                                               
/usr/local/lib/pulp/lib64/python3.9/site-packages/pulpcore/client/pulp_ansible/api/pulp_ansible_galaxy_api_v3_versions_api.py:265: in read                                                    
    return self.read_with_http_info(name, namespace, path, version, **kwargs)  # noqa: E501
/usr/local/lib/pulp/lib64/python3.9/site-packages/pulpcore/client/pulp_ansible/api/pulp_ansible_galaxy_api_v3_versions_api.py:372: in read_with_http_info
    return self.api_client.call_api(
/usr/local/lib/pulp/lib64/python3.9/site-packages/pulpcore/client/pulp_ansible/api_client.py:365: in call_api
    return self.__call_api(resource_path, method,
/usr/local/lib/pulp/lib64/python3.9/site-packages/pulpcore/client/pulp_ansible/api_client.py:209: in __call_api
    return_data = self.deserialize(response_data, response_type)
/usr/local/lib/pulp/lib64/python3.9/site-packages/pulpcore/client/pulp_ansible/api_client.py:281: in deserialize
    return self.__deserialize(data, response_type)
/usr/local/lib/pulp/lib64/python3.9/site-packages/pulpcore/client/pulp_ansible/api_client.py:320: in __deserialize
    return self.__deserialize_model(data, klass)
/usr/local/lib/pulp/lib64/python3.9/site-packages/pulpcore/client/pulp_ansible/api_client.py:659: in __deserialize_model
    kwargs[attr] = self.__deserialize(value, attr_type)
/usr/local/lib/pulp/lib64/python3.9/site-packages/pulpcore/client/pulp_ansible/api_client.py:320: in __deserialize
    return self.__deserialize_model(data, klass)
/usr/local/lib/pulp/lib64/python3.9/site-packages/pulpcore/client/pulp_ansible/api_client.py:659: in __deserialize_model
    kwargs[attr] = self.__deserialize(value, attr_type)
/usr/local/lib/pulp/lib64/python3.9/site-packages/pulpcore/client/pulp_ansible/api_client.py:303: in __deserialize
    for k, v in six.iteritems(data)}
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

d = [], kw = {}                                

    def iteritems(d, **kw):
>       return iter(d.items(**kw))
E       AttributeError: 'list' object has no attribute 'items'

/usr/local/lib/pulp/lib64/python3.9/site-packages/six.py:589: AttributeError

```

I found this issue: https://github.com/tfranzel/drf-spectacular/issues/238
and I discovered there is a setting for this, I'll use that, thanks for your comment!

Comment (mdellweg) [reply pulpcore/openapi/__init__.py]: So we can have one less patch? That is great!

**Review by mdellweg — APPROVED** [review]
This is great!

**Review by dralley — APPROVED** [review]

**Comment by fao89** [comment]
this can be merged now

**Pull request merged**